enhancements/installer/log-config.md

```diff
@@ -0,0 +1,189 @@
+configuring logging behavior in the Installer.
+
+## Motivation
+
+* Decorating installer logs with k/v pairs provided by the ARO resource provider is the
+top priority ask for the installer in the 4.12 release. The ARO team wants this feature
+in order to filter logs when logs from multiple installs are pooled. For example, all installer
+logs could be decorated with a k/v pair denoting an aroClusterID:
+
+```shell
+INFO OpenShift Installer unreleased-master-6230-gf46ac2f0d1a3628a51d3437bd79e4221da9aff23-dirty  aroClusterID=123
+INFO Built from commit f46ac2f0d1a3628a51d3437bd79e4221da9aff23  aroClusterID=123
+INFO Fetching Master Machines...                   aroClusterID=123
+INFO Loading Master Machines...                    aroClusterID=123
+```
+
+* We can generalize the ask from ARO so that it may be useful to other users
+* We should keep future logging needs in mind when designing this solution 
+so that we do not complicate or preclude future development work
+
+
+### User Stories
+
+* As an installer user, I want to be able to provide a set of key-value pairs that
+will decorate all Installer log lines
+* As an installer user, I would like to configure the level of logging through a config file 
+* As an installer developer, I want the design for this user input to be extensible
+
+### Goals
+
+The goal of this enhancement is to determine the proper way of passing user input
+for the need of configuring logging.
+
+### Non-Goals
+
+The need for decorating log lines is just an example of the kind of configuration
+that is needed. The implementation for decorating log lines is not relevant
+to this discussion.
+
+## Proposal
+
+This design proposes we add support for a log-config.yaml file to
+allow users to pass in logging configuration, for example:
+
+```go
+type LogConfig struct {
+  // Fields contains user-provided fields to be added to all log entries.
+  // +optional
+  Fields map[string]string `json:"fields,omitempty"`
+
+  // Level sets the level of logging to standard out. 
+  // Valid values: Info, Debug, Warn, Error
+  // +optional
+  Level string `json:"level,omitempty"`
+}
+```
+
+The install-config is the typical vehicle for installer configuration,
+but it is not acceptable in this use case because the install-config
+is an Installer Asset, and logging must be configured:
+
+1. Before assets are initialized
+2. Despite asset failures
+
+In the example use case of decorating logs with fields, we need
+to decorate lines that are output before the install config is loaded
+and we need to decorate output that would occur if there are
+errors in the install config and it cannot be loaded.
+
+The log-config.yaml file would be loaded if present in the install directory,
+as well as adding a command line flag  --log-config  to accept a path to the config file.
+
+Users should be able to provide their config in a file that looks, for example, like:
+
+```yaml
+fields:
+  key1: value1
+  key2: value2
+  key3: value3
+level: debug
```

Comment:
Can you clarify the meaning of level here? From the description above "Level sets the level of logging to standard out" I assumed it was setting the global log level.
But https://github.com/openshift/installer/pull/6161 states that it's specifying which log levels the extra keys will be applied to.

Follow-up question: is this actually useful?

Reply:
Yes, this needs to be clarified. `level` isn't really a main priority for this enhancement and that needs to be made more clear. It's probably a distraction and should be removed entirely, but it was intended as an example or nice-to-have analogue for  setting the current log levels already configured by `--log-level`

Reply:
FWIW we don't intend to use it from hive.